avr/bootloaders/HoodLoader2/makefile

```diff
@@ -23,6 +25,15 @@ LUFA_PATH    = ../lufa-LUFA-140928/LUFA
 CC_FLAGS     = -DUSE_LUFA_CONFIG_HEADER -IConfig/ $(HOODLOADER2_OPTS) -DBOOT_START_ADDR=$(BOOT_START_OFFSET) $(REGS) -flto -fuse-linker-plugin
 LD_FLAGS     = -Wl,--section-start=.text=$(BOOT_START_OFFSET) $(BOOT_API_LD_FLAGS),--section-start=.data=$(RAM_OFFSET) $(REGS) -flto -fuse-linker-plugin
 
+ifneq ($(MCU),atmega8u2)
+ifneq ($(MCU),atmega16u2)
+ifneq ($(MCU),atmega16u4)
+ifneq ($(MCU),atmega32u4)
+$(error MCU must be one of atmega8u2, atmega16u2, atmega16u4 or atmega32u4)
```

Comment:
I think this is not required as the build would just fail. Also you missed the at90usb devices.